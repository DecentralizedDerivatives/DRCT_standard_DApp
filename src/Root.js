import React from 'react';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import { createLogger } from 'redux-logger';
import rootReducer from './reducers';

const middlewares = [thunk];
let composeEnhancer = compose;

if (process.env.NODE_ENV === 'development') {
  middlewares.push(createLogger());
  composeEnhancer = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
}

const Root = ({ children, initialState = {} }) => {
  const store = createStore(
    rootReducer,
    initialState,
    composeEnhancer(applyMiddleware(...middlewares))
  );

  return <Provider store={store}>{children}</Provider>;
};

export default Root;
