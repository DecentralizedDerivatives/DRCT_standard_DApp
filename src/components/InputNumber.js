import React from 'react';
import { InputGroupAddon } from 'reactstrap'
const InputNumber = ({ input, label, className, addonLabel, meta: { touched, error } }) => (
  <div className="input-number">
    <label htmlFor={input.name}>{label}</label>
    <input className={className} {...input} type="number" step=".00001" min="0" />
    {addonLabel&& (
      <InputGroupAddon addonType="append">{addonLabel}</InputGroupAddon>
    )}
    {touched && (error && <span className="input-error">{error}</span>)}
  </div>
);

export default InputNumber;