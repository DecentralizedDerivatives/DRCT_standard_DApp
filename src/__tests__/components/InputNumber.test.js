import React from 'react';
import { shallow } from 'enzyme';
import { initStore } from '../../Root';

import InputNumber from '../../components/InputNumber';

describe('<InputNumber />', () => {
  describe('render()', () => {
    it('renders the component', () => {
      const wrapper = shallow(
        <InputNumber store={initStore()} meta={{}} input={{}} />
      );
      expect(wrapper).toMatchSnapshot();
    });
  });
});
