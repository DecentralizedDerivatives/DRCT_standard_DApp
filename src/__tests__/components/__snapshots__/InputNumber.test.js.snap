// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputNumber /> render() renders the component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InputNumber
    input={Object {}}
    meta={Object {}}
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label
          className="input-label"
        />,
        <input
          min="0"
          step=".00001"
          type="number"
        />,
        undefined,
        undefined,
      ],
      "className": "input-number",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "input-label",
          "htmlFor": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": undefined,
          "min": "0",
          "step": ".00001",
          "type": "number",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      undefined,
      undefined,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label
            className="input-label"
          />,
          <input
            min="0"
            step=".00001"
            type="number"
          />,
          undefined,
          undefined,
        ],
        "className": "input-number",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "input-label",
            "htmlFor": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": undefined,
            "min": "0",
            "step": ".00001",
            "type": "number",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        undefined,
        undefined,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
